src/Microsoft.Azure.WebJobs.Extensions.EventHubs/Listeners/EventHubListener.cs

```diff
@@ -315,13 +315,15 @@ private bool TryGetLinkedActivity(EventData message, out Activity link)
             {
                 link = null;
 
-                if (!message.Properties.ContainsKey("Diagnostic-Id"))
+                if ((message.SystemProperties.TryGetValue("Diagnostic-Id", out var diagnosticIdObj) || message.Properties.TryGetValue("Diagnostic-Id", out diagnosticIdObj)) 
+                    && diagnosticIdObj is string diagnosticIdString)
                 {
-                    return false;
+                    link = new Activity("Microsoft.Azure.EventHubs.Process");
+                    link.SetParentId(diagnosticIdString);
+                    return true;
                 }
 
-                link = message.ExtractActivity();
```

Comment:
Do we not need to call `ExtractActivity` anymore?

Reply:
ExtractActivity is only available on Application-Properties, and changing old EventHub SDK does not seem reasonable option  as it's in the maintenance mode and Azure SDK is shipping [new flavor](https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/eventhub/Azure.Messaging.EventHubs).

 I took the only part of `ExtractActivity` we care about (which is `activitiy.SetParentId`). Besides that  `ExtractActivity` also populates Baggage on the Activity which we drop for EventHub scenario anyway.